pkg/authz/warrant/mysql.go

```diff
@@ -384,7 +384,45 @@ func (repo MySQLRepository) GetAllMatchingObjectAndRelation(ctx context.Context,
 			FROM warrant
 			WHERE
 				objectType = ? AND
-				objectId = ? AND
+				(objectId = ? OR objectId = "*") AND
+				relation = ? AND
+				(contextHash = ? OR contextHash = "") AND
+				deletedAt IS NULL
+			ORDER BY createdAt DESC, id DESC
+		`,
+		objectType,
+		objectId,
+		relation,
+		contextHash,
+	)
+	if err != nil {
+		switch err {
+		case sql.ErrNoRows:
+			return models, nil
+		default:
+			return nil, errors.Wrapf(err, "error getting warrants with object type %s, object id %s, and relation %s", objectType, objectId, relation)
+		}
+	}
+
+	for i := range warrants {
+		models = append(models, &warrants[i])
+	}
+
+	return models, nil
+}
+
+func (repo MySQLRepository) GetAllMatchingObjectAndRelationWithSubjectType(ctx context.Context, objectType string, objectId string, relation string, subjectType string, contextHash string) ([]Model, error) {
```

Comment:
I think a slightly better name might be `GetAllMatchingObjectAndRelationBySubjectType`. Be sure to apply this change to the postgres and sqlite repositories as well.